pylib/gyp/common.py

```diff
@@ -422,17 +422,62 @@ def EnsureDirExists(path):
     except OSError:
         pass
 
+def GetCrossCompilerPredefines():
+    CC = os.environ.get("CC_target") or os.environ.get("CC")
+    CFLAGS = os.environ.get("CFLAGS")
+    CXX = os.environ.get("CXX_target") or os.environ.get("CXX")
+    CXXFLAGS = os.environ.get("CXXFLAGS")
+    cmd = []
+    defines = {}
+
+    if CC:
+        cmd += CC.split(" ")
+        if CFLAGS:
+            cmd += CFLAGS.split(" ")
+    elif CXX:
+        cmd += CXX.split(" ")
+        if CXXFLAGS:
+            cmd += CXXFLAGS.split(" ")
+    else:
+        return defines
```

Comment:
```suggestion
    cmd
```

Reply:
GetCrossCompilerPredefines should return `defines (dict)`, not the `cmd`

Reply:
> defines will ALWAYS be empty.

No, `defines` is filled in

```py
    try:
        out = subprocess.Popen(
            [*cmd, "-dM", "-E", "-x", "c", input],
            shell=sys.platform == "win32",
            stdout=subprocess.PIPE, stderr=subprocess.STDOUT
        )
        lines = out.communicate()[0].decode("utf-8").replace(
            "\r\n", "\n").split("\n")
        for line in lines:
            if not line:
                continue
            define_directive, key, *value = line.split(" ")
            assert define_directive == "#define"
            defines[key] = " ".join(value)  # <---------
        return defines  # <---------
    finally:
        if sys.platform == "win32":
            os.unlink(input)
```

Reply:
> cmd may be filled up but it will ALWAYS be thrown away.

and cmd is used here

```py
try:
        out = subprocess.Popen(
            [*cmd, "-dM", "-E", "-x", "c", input], # <---------
            shell=sys.platform == "win32",
            stdout=subprocess.PIPE, stderr=subprocess.STDOUT
        )
        lines = out.communicate()[0].decode("utf-8").replace(
            "\r\n", "\n").split("\n")
        for line in lines:
            if not line:
                continue
            define_directive, key, *value = line.split(" ")
            assert define_directive == "#define"
            defines[key] = " ".join(value)
        return defines
    finally:
        if sys.platform == "win32":
            os.unlink(input)
```